packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+![SLE Files grid](../spec-images/fileSizesSle.png)
+
+Windows explorer (details view)
+
+![Windows explorer](../spec-images/fileSizesWindows.png)
+
+Chrome Dev Tools Network tab
+
+![Chrome Dev Tools Network tab](../spec-images/fileSizeChromeDevTools.png)
+
+---
+
+## Design
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
+
+The component will extend `TableColumnTextBase`, thereby including the APIs for a **groupable** and **fractional-width** column.
```

Comment:
To confirm, do we think it makes sense for this column to be groupable? That doesn't seem particularly useful unless we think there's a use case for this column that would only have a discrete set of file sizes.

Reply:
That's a fair point. I wonder, though, is it worth removing the ability to group from this column, when every (?) other column type allows it? I guess I very slightly lean towards making it non-groupable.

Reply:
We had an out-of-thread discussion regarding limiting grouping in the API and decided that it was sufficient for clients to simply leverage the [`groupingDisabled` API](https://github.com/ni/nimble/blob/990e415d40fa3abe36414c1736b2133f7cd0cd03/packages/nimble-components/src/table-column/mixins/groupable-column.ts#L20C16-L20C33) if they wanted to prevent users from grouping by this column.

I could imagine a scenario where a client wanted to use this column to represent products that had a very discrete set of values for byte size (i.e. hard drives or memory sticks), and it would make a ton of sense for those values to be grouped by.